pbiapi/pbiapi.py

```diff
@@ -1,400 +1,327 @@
         response = requests.get(url, headers=self.headers)
-        
-        if response.status_code != 200:
-            return False
+
+        if response.status_code != HTTP_OK_CODE:
+            print(f"Failed when checking if the workspace, '{name}' already exists!")
+            self.force_raise_http_error(response)
+
+        if response.json()["@odata.count"] > 0:
+            print("Workspace already exists, no changes made!")
+            return
+
+        # Workspace does not exist, lets create it:
+        print(f"Trying to create a workspace with name: {name}...")
+        url = self.base_url + "groups?workspaceV2=true"
+        response = requests.post(url, data={"name": name}, headers=self.headers)
+
+        if response.status_code == HTTP_OK_CODE:
+            print("Workspace created successfully!")
+            self.get_workspaces()  # Update internal state
         else:
-            #Dont reproduce workspace
-            if response.json()['@odata.count'] > 0:
-                print('Dataset already exists')
-                return False
-            #Try to create workspace
-            else:
-                print('Creating a workspace')
-                url = "https://api.powerbi.com/v1.0/myorg/groups?workspaceV2=true"
-                payload = {
-                    "name":name
-                }
-                response = requests.post(url, data=payload, headers=self.headers)
-
-                if response.status_code == 200:
-                    print("Created workspace: ",name)
-                    self.getWorkspaces()
-                    return True
-                else:
-                    print("Error Creating workspace")
-                    print(response.text)
-                    return False
-        
-    @checkToken
-    def addUsersToWorkspace(self,name,users):
-        self.getWorkspaces()
-        workspaceId = self.findWorkspaceIdByName(name)
-
-        if workspaceId:
-            url = "https://api.powerbi.com/v1.0/myorg/groups/{groupId}/users".format(groupId = workspaceId)
-            
-            response = requests.post(url, data=users, headers=self.headers)
-            if response.status_code == 200:
-                print("Added users to workspace")
-                return True
-            else:
-                print("Error adding users to workspace")
-                print(response.text)
-                return False
+            print(f"Failed to create the new workspace: '{name}':")
+            self.force_raise_http_error(response)
+
+    @check_token
+    def add_users_to_workspace(self, workspace_name, users):
+        self.get_workspaces()
+        workspace_id = self.find_workspace_id_by_name(workspace_name, raise_if_missing=True)
+
+        # Workspace exists, lets add users:
+        url = self.base_url + f"groups/{workspace_id}/users"
+        response = requests.post(url, data=users, headers=self.headers)
```

Comment:
Regards from Anton: According to the API docs only a single user is being added by this endpoint. Could this be verified? If so, this method should be renamed (users -> user).

Reply:
Yes, given that you use the same endpoint to add a user, a group or an app (Service Principal) this could be named better. I'm up for suggestions!